hudi-spark-datasource/hudi-spark/src/main/java/org/apache/hudi/cli/BootstrapExecutorUtils.java

```diff
@@ -263,10 +262,9 @@ private void initializeTable() throws IOException {
 
   private Pair<String, String> genKeyGenClassAndPartitionColumns() {
     String keyGenClass;
-    if (StringUtils.nonEmpty(props.getString(HoodieBootstrapConfig.KEYGEN_CLASS_NAME.key(), null))) {
-      keyGenClass = props.getString(HoodieBootstrapConfig.KEYGEN_CLASS_NAME.key());
-    } else if (StringUtils.nonEmpty(props.getString(HoodieBootstrapConfig.KEYGEN_TYPE.key(), null))) {
-      props.put(HoodieWriteConfig.KEYGENERATOR_TYPE.key(), props.getString(HoodieBootstrapConfig.KEYGEN_TYPE.key()));
+    if (StringUtils.nonEmpty(props.getString(HoodieWriteConfig.KEYGENERATOR_CLASS_NAME.key(), null))) {
```

Comment:
how does backwards compatability work? we are not going to support older configs? and ask users to set write configs instead?

Reply:
Since the bootstrap operation is an one-time operation, it is ok to ask user to use new configs and we'll mention in the docs that the old configs are removed.  Also, in common use cases where simple keygen is used, such a config is not required.